rust/src/utils/distance.rs

```diff
@@ -26,6 +26,18 @@ use arrow_schema::DataType;
 
 use crate::Result;
 
+#[inline]
+pub(crate) fn simd_alignment() -> i32 {
+    #[cfg(any(target_arch = "x86_64"))]
+    {
```

Comment:
this should be `AVX256` -> 8, `AVX512` -> 16, tho, and `neon` -> 4. 

We can change it later if we decide to add AVX512 support